jsparagus/grammar.py

```diff
@@ -595,6 +610,16 @@ def validate_nt(nt, nt_def):
                     (), [Production([goal], 0), Production([init_nt, End()], 'accept')], types.NoReturnType)
             self.init_nts.append(init_nt)
 
+    def patch(self, extensions):
+        if extensions == []:
+            return
+        # Copy of nonterminals which would be mutated by the patches.
+        nonterminals = { nt: nt_def for nt, nt_def in self.nonterminals.items() }
```

Comment:
you can use `copy.copy` for shallow copy of a dict
https://docs.python.org/3.7/library/copy.html